crates/api/src/site.rs

```diff
@@ -213,13 +227,19 @@ impl Perform for Search {
         .await??;
       }
       SearchType::All => {
+        // If the community or creator is included, dont search communities or users
+        let community_or_creator_included =
+          data.community_id.is_some() || data.community_name.is_some() || data.creator_id.is_some();
```

Comment:
This is necessary because there is no creator or community filter for a few search types.